services/order.py

```diff
@@ -0,0 +1,42 @@
+from datetime import datetime
+
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+from db.models import Ticket, Order
+
+
+@transaction.atomic
+def create_order(
+    tickets: list[dict[str, int]],
+    username: str,
+    date: datetime = None
+) -> None:
+    user = get_user_model().objects.get(username=username)
+    order = Order.objects.create(user=user)
+
+    if date:
+        order.created_at = date
+        order.save()
```

Comment:
Directly setting the `created_at` field after the order is created might not be necessary, as `auto_now_add=True` should automatically set this field when the order is created. If you need to set a custom date, ensure this does not conflict with any business logic or auditing requirements.